http/codegen/service_data.go

```diff
@@ -1502,20 +1553,31 @@ func buildBodyType(svc *service.Data, s *httpdesign.ServiceExpr, e *httpdesign.E
 			rctx = "result"
 		}
 		desc = fmt.Sprintf("%s builds the HTTP %s body from the %s of the %q endpoint of the %q service.",
-			name, ctx, rctx, e.Name(), s.Name())
+			name, ctx, rctx, e.Name(), svc.Name)
 
-		// If design uses Body("name") syntax then need to use payload
-		// attribute to transform.
 		srcType := att.Type
+		tgtType := body.Type
+		if svr && !req {
+			// In server response type all the attributes are pointers. So we need to
+			// remove the required validations from the body to transform.
```

Comment:
Could we do it so that `body.Type` is correctly initialized in that case? or would that break something else? if so please explain in comment.